src/pybind/mgr/dashboard/frontend/src/app/shared/services/notification.service.ts

```diff
@@ -123,10 +108,39 @@ export class NotificationService {
           application
         );
       }
+      this.queueToShow(config);
+    }, 10);
+  }
+
+  private queueToShow(config: CdNotificationConfig) {
```

Comment:
I would suggest the following change:
```diff
diff --git a/src/pybind/mgr/dashboard/frontend/src/app/shared/services/notification.service.ts b/src/pybind/mgr/dashboard/frontend/src/app/shared/services/notification.service.ts
index 5ea418d9cb..d90541c121 100644
--- a/src/pybind/mgr/dashboard/frontend/src/app/shared/services/notification.service.ts
+++ b/src/pybind/mgr/dashboard/frontend/src/app/shared/services/notification.service.ts
@@ -124,13 +124,14 @@ export class NotificationService {
 
   private showQueued() {
     const byTitle = {};
-    this.queued.forEach((n) => {
+    while (this.queued.length > 0) {
+      const n = this.queued.shift();
       if (!byTitle[n.title]) {
         byTitle[n.title] = [n];
       } else {
         byTitle[n.title].push(n);
       }
-    });
+    }
     Object.values(byTitle).forEach((configs: CdNotificationConfig[]) => {
       const config = configs[0];
       if (configs.length > 1) {
@@ -140,7 +141,6 @@ export class NotificationService {
       this.save(notification);
       this.showToasty(notification);
     });
-    this.queued = [];
   }
 
   private showToasty(notification: CdNotification) {
```
With the current code there is a really **small** chance that a notification is removed without being displayed.
If a notification is added while the browser is running lines 134 and 142, it will be removed in line 143 and it is never read.

Reply:
nice spotting :)
Using shift now